src/l3.c

```diff
@@ -112,9 +134,15 @@ l3__log_simple(uint32_t loc, const char *msg, const uint64_t arg1, const uint64_
 {
     int idx = __sync_fetch_and_add(&l3_log->idx, 1) % L3_MAX_SLOTS;
     l3_log->slots[idx].tid = l3_mytid();
+
 #ifdef L3_LOC_ENABLED
     l3_log->slots[idx].loc = (loc_t) loc;
+#else
+#ifdef DEBUG
+    assert(l3_log->slots[idx].loc == 0);
```

Comment:
IF `L3_LOC_ENABLED` is false (OFF), we now will no longer set `loc = 0` (to avoid minor ns-overhead).

Instead, under `DEBUG` mode , assert() that this field is supposed to be 0.

The Py dump script relies on this field being 0. Otherwsie, in future-revs, that script will try to unpack the contents of a non-zero `loc` field, which, if it's unitialized, will result in returning bogus code-location information.